changelogs/unreleased/6058-flawedmatrix-minor.md

```diff
@@ -0,0 +1,3 @@
+## Allow setting connection limit per listener
+
+Adds a `max-connections-per-listener` config option to limit the number of active connections to a listener. The default, if unset, is unlimited.
```

Comment:
It'd be nice to spell out the full YAML path of the config option for both the ContourConfiguration CRD (e.g. `spec.envoy.listener.maxConnectionsPerListener`) and the Contour config file (`listener.max-connections-per-listener`) here for clarity.